File: 14_ BDD (Behavior Driven Development)/Praktikum/Eksplorasi-BDD/cypress/e2e/payment/payment.js
import {Given , When, And, Then} from "cypress-cucumber-preprocessor/steps";

Given ("I am on the payment pages",function(){
    console.log("Payment Page")
});
When ("I select the payment method",function(){
    console.log("Payment method verified")
});
And ("I click the Pay Now button",function(){
    console.log("Pay Now verified")
});
Then ("I am on the payment confirmation page",function(){
    console.log("Payment confirmation verified")
});
When ("I click Payment Instructions with Scan",function(){
    console.log("Payment instruction verified")
});
And ("I open the Gojek application then head to the GoPay menu",function(){
    console.log("Applicattion open verified")
});
Then ("I select the Pay feature",function(){
    console.log("Pay feature verified")
});
When ("I do a scan of the payment",function(){
    console.log("Scan payment verified")
});
And ("I enter your 6-digit GoPay PIN",function(){
    console.log("Digit Pin verified")
});
Then ("I am on the paid success page",function(){
    console.log("Paid success verified")
});

File: 14_ BDD (Behavior Driven Development)/Praktikum/BDD-Prioritas2/src/test/java/starter/payment/Payment.java
package starter.payment;

import net.thucydides.core.annotations.Step;

public class Payment {
    @Step("I am on the payment pages")
    public void onThePaymentPages(){
        System.out.println("I am on the payment pages");
    }

    @Step("I select the payment method")
    public void selectThePaymentMethod(){
        System.out.println("I select the payment method");
    }

    @Step("I click the Pay Now button")
    public void clickPayNowButton(){
        System.out.println("I click the Pay Now button");
    }

    @Step("I am on the payment confirmation page")
    public void onThePaymentConfirmationPage(){
        System.out.println("I am on the payment confirmation page");
    }
    @Step("I click Payment Instructions with Scan")
    public void clickPaymentInstructions(){
        System.out.println("I click Payment Instructions with Scan");
    }

    @Step("I open the Gojek application then head to the GoPay menu")
    public void openApplication(){
        System.out.println("I open the Gojek application then head to the GoPay menu");
    }

    @Step("I select the Pay feature")
    public void selectPayFeature(){
        System.out.println("I select the Pay feature");
    }

    @Step("I do a scan of the payment")
    public void doScan(){
        System.out.println("I do a scan of the payment");
    }

    @Step("I enter your 6-digit GoPay PIN")
    public void enterYourDigit(){
        System.out.println("I enter your 6-digit GoPay PIN");
    }

    @Step("I am on the paid success page")
    public void onThePaidSuccessPage(){
        System.out.println("I am on the paid success page");
    }
}
